tests/test_repeating.js

```diff
@@ -1,90 +1,94 @@
 'use strict'
 
-const http = require('http')
-const async = require('async')
 const { test } = require('tap')
 const nock = require('..')
 const got = require('./got_client')
+const assertRejects = require('assert-rejects')
 
 require('./cleanup_after_each')()
 
-test('repeating once', t => {
-  nock.disableNetConnect()
-
-  nock('http://example.test')
+test('repeating once', async t => {
+  const scope = nock('http://example.test')
     .get('/')
     .once()
     .reply(200, 'Hello World!')
 
-  http.get('http://example.test', function(res) {
-    t.equal(200, res.statusCode, 'first request')
-    t.end()
-  })
-})
+  const { statusCode } = await got('http://example.test/')
+  t.is(statusCode, 200)
 
-test('repeating twice', t => {
-  nock.disableNetConnect()
+  await assertRejects(
+    got('http://example.test/'),
+    Error,
+    'Nock: No match for request'
+  )
 
-  nock('http://example.test')
+  scope.done()
+})
+
+test('repeating twice', async t => {
+  const scope = nock('http://example.test')
     .get('/')
     .twice()
     .reply(200, 'Hello World!')
 
-  async.each(
-    [1, 2],
-    function(_, cb) {
-      http.get('http://example.test', function(res) {
-        t.equal(200, res.statusCode)
-        cb()
-      })
-    },
-    t.end.bind(t)
+  // eslint-disable-next-line no-unused-vars
+  for (const _ of Array(2)) {
```

Comment:
This pattern could be replaced by LoDash's [`times`](https://lodash.com/docs/4.17.4#times) util. 
Not sure if it would make it easier to follow.
```js
_.times(2, () => {
  const { statusCode } = await got('http://example.test/')
  t.is(statusCode, 200)
})
```

Reply:
I’d prefer to run the async operations in series and don’t think the lodash version will do that. Also I’m hoping to move away from lodash except if it’s really needed.

Reply:
[lodash.times](https://github.com/lodash/lodash/blob/master/times.js) does run in series, but I understand not wanting to use a lib here.

Reply:
It runs synchronous functions in series, though not functions that return promises. The effect is that when there is an `await` in the iteratee, it doesn't wait for it to finish before kicking off the next one.

Reply:
ah right. I wrote out that snippet without making the callback async, but that's not valid. 😊